example-nextjs/components/CodeRunnerUI.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useRef } from 'react'
+import Editor from '@monaco-editor/react'
+import Script from 'next/script'
+import GithubButton from './GithubButton'
+import Navbar from './Navbar'
+
+interface Props {
+  initialCode: string
+  onRunCode(inputCode: string): Promise<string | void>
+  languageLabel: string
+  hideRunCode?: boolean
+  children?: React.ReactNode
+  hideOutputEditor?: boolean
+}
+
+export default function CodeRunnerUI(props: Props) {
```

Comment:
nit: why if we destructure `props` so you can write `initialCode` instead of `props.initialCode`?

Reply:
Destructured in the next line.